editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2020 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.edit_with_branch_versioning;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.TextField;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.VBox;
+import javafx.geometry.Point2D;
+
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionInfo;
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionParameters;
+import com.esri.arcgisruntime.arcgisservices.VersionAccess;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.ArcGISFeature;
+import com.esri.arcgisruntime.data.FeatureTableEditResult;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.data.ServiceGeodatabase;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.GeoElement;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+
+public class EditWithBranchVersioningController {
+
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private Button createVersionButton;
+  @FXML private Button switchVersionButton;
+  @FXML private VBox editFeatureVBox;
+  @FXML private ComboBox<VersionAccess> accessComboBox;
+  @FXML private ComboBox<String> damageTypeComboBox;
+  @FXML private Label currentVersionLabel;
+  @FXML private VBox createVersionVBox;
+  @FXML private TextField name;
+  @FXML private TextField description;
```

Comment:
Might also be worth alphabetising these?
```suggestion
  @FXML private Button createVersionButton;
  @FXML private Button switchVersionButton;
  @FXML private ComboBox<String> damageTypeComboBox;
  @FXML private ComboBox<VersionAccess> accessComboBox;
  @FXML private Label currentVersionLabel;
  @FXML private MapView mapView;
  @FXML private ProgressIndicator progressIndicator;
  @FXML private TextField description;
  @FXML private TextField name;
  @FXML private VBox createVersionVBox;
  @FXML private VBox editFeatureVBox;
```